src/modules/blockchain/implementation/web3-service.js

```diff
@@ -49,22 +52,52 @@ class Web3Service {
     async initializeContracts() {
         this.hubContract = new this.web3.eth.Contract(Hub.abi, this.config.hubContractAddress);
 
-        const UAIRegistryAddress = await this.callContractFunction(
+        const assetRegistryAddress = await this.callContractFunction(
             this.hubContract,
             'getContractAddress',
-            ['UAIRegistry'],
+            ['AssetRegistry'],
         );
-        this.UAIRegistryContract = new this.web3.eth.Contract(UAIRegistry.abi, UAIRegistryAddress);
+        this.AssetRegistryContract = new this.web3.eth.Contract(AssetRegistry.abi, assetRegistryAddress);
+
+        const tokenAddress = await this.callContractFunction(
+            this.hubContract,
+            'getContractAddress',
+            ['Token'],
+        );
+        this.TokenContract = new this.web3.eth.Contract(ERC20Token.abi, tokenAddress);
```

Comment:
Can we wrap web3 calls with our function. It will be easier to support multiple RPC endpoints in the future

Reply:
Could you give an example? I am already using execute/call contract functions

Reply:
new this.web3.eth.Contract(ERC20Token.abi, tokenAddress); can be something like:
 getContractObject(abi, address) {
return new this.web3.eth.Contract(ERC20Token.abi, tokenAddress); 
}